utils/docker/Dockerfile.centos.7

```diff
@@ -12,13 +12,44 @@ LABEL maintainer="daos@daos.groups.io"
 # Intermittent cache-bust.  Used to reduce load on the actual CACHEBUST later.
 ARG CB0
 
+# Need to be able to use a local repository for packages
+ARG REPO_URL=""
+ARG REPO_EL7=""
+
+# Install EPEL and switch to dnf as it seems a bit faster.
+RUN yum clean all &&                                            \
+    yum -y install dnf epel-release yum-utils &&                \
+    yum clean all
+
+# If a local repository is supplied, then we should use it
+# and disable using the distro repositories.
+RUN if [ -n "$REPO_URL" ]; then                                 \
+      MY_REPO="${REPO_URL}${REPO_EL7}/";                        \
+      MY_NAME="${MY_REPO#*//}";                                 \
+      MY_NAME="${MY_NAME//\//_}";                               \
+      echo -e "[${MY_NAME}]\n\
+name=created from ${MY_REPO}\n\
+baseurl=${MY_REPO}\n\
+enabled=1\n\
+repo_gpgcheck=0\n\
+exclude=fuse-3* fuse-devel-3* fuse-libs-3* fuse-debuginfo-3* ompi*\n\
+gpgcheck=0\n" >> /etc/yum.repos.d/local-centos-group.repo;      \
```

Comment:
If this repo contains RPMs that we fetch from upstream mirrors, we **MUST NOT** be disabling the GPG checking on it.

We **MUST** be GPG checking any RPM that we do not produce our selves.  This may mean we will have to either be signing our own packages on every build (which then becomes a prerequisite to being able to do this) or be keeping the upstream mirror repos separate from our local repos which have unsigned RPMs in them (but that we can trust, because we produced them, didn't download them from the Internet-at-large.

Reply:
That is going to take another change to pipeline lib before I can fix it in this PR.

The change will be that there will be one per distro docker specific group repos with package GPG check enabled and REPO gpg check disabled.

And then a one of stable|release|dev repos for locally built artifacts with all gpg checks disabled.